### ISSUE TITLE
'Second-instance' fires instead of 'open-url' in electron on mac

### ISSUE THREAD

**Issue by rbudnar** [electron/electron#20088]
### Preflight Checklist

* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

* **Electron Version:**
Electron `6.0.7`, `5.0.1`
* **Operating System:**
`MacOS Mojave 10.14.6`
* **Last Known Working Electron version:**
  `n/a`

### Expected Behavior

See this SO post for a description of the issue: https://stackoverflow.com/questions/57733424/second-instance-fires-instead-of-open-url-in-electron-on-mac

I would expect that opening an electron app from the finder/launchpad in MacOS should fire the `open-url` event when using a protocol link to open the app from the web, e.g., `myapp://`. If not, I would expect the protocol URL to somehow be available in the `second-instance` event as it is on windows.

The main problem that I need to resolve is that I need to parse/use the parameters that I will be embedding in the `myapp://` link (e.g., `myapp://param1=123`), which does not appear to be available from the `second-instance` event on mac. 

### Actual Behavior

When you open the app from the finder/launch pad, the `second-instance` event will fire when using the protocol link.

### To Reproduce
1. Clone the `electron-quick-start` repo, add `electron-packager` as a dev dependency, and then npm install
2. Set up app to register as default protocol for app:
update package.json:
```
{
...
"protocols": {
    "name": "myapp",
    "schemes": [
      "myapp"
    ]
  }
...
}
```
update main.js to add in the single instance lock and protocol registration:
<details>

```
// Modules to control application life and create native browser window
const {app, BrowserWindow} = require('electron')
const path = require('path')

// Keep a global reference of the window object, if you don't, the window will
// be closed automatically when the JavaScript object is garbage collected.
let mainWindow

function logEverywhere(s) {
  console.log(s)
  if (mainWindow && mainWindow.webContents) {
      mainWindow.webContents.executeJavaScript(`console.log("${s}")`)
  }
}

function createWindow () {
  // Create the browser window.
  mainWindow = new BrowserWindow({
    width: 800,
    height: 600,
    webPreferences: {
      preload: path.join(__dirname, 'preload.js')
    }
  })

  // and load the index.html of the app.
  mainWindow.loadFile('index.html')

  mainWindow.webContents.openDevTools({ mode: "detach" });
  // Open the DevTools.
  // mainWindow.webContents.openDevTools()

  // Emitted when the window is closed.
  mainWindow.on('closed', function () {
    // Dereference the window object, usually you would store windows
    // in an array if your app supports multi windows, this is the time
    // when you should delete the corresponding element.
    mainWindow = null
  })
}

const gotTheLock = app.requestSingleInstanceLock();

if (!gotTheLock) {
    app.quit();
} else { 
  app.on('second-instance', (event, commandLine, workingDirectory) => {
    // Someone tried to run a second instance, we should focus our window.
    if (mainWindow) {
      if (mainWindow.isMinimized()) mainWindow.restore()
      mainWindow.focus()
      logEverywhere("second instance");
    }
  });
}

app.setAsDefaultProtocolClient("myapp");

app.on("will-finish-launching", () => {
  app.on("open-url", (event, url) => {
      event.preventDefault();
      logEverywhere(`open-url inside wfl: ${url}`);
  });
});

// This method will be called when Electron has finished
// initialization and is ready to create browser windows.
// Some APIs can only be used after this event occurs.
app.on('ready', createWindow)

// Quit when all windows are closed.
app.on('window-all-closed', function () {
  // On macOS it is common for applications and their menu bar
  // to stay active until the user quits explicitly with Cmd + Q
  if (process.platform !== 'darwin') app.quit()
})

app.on('activate', function () {
  // On macOS it's common to re-create a window in the app when the
  // dock icon is clicked and there are no other windows open.
  if (mainWindow === null) createWindow()
})

// In this file you can include the rest of your app's specific main process
// code. You can also put them in separate files and require them here.
```
</details>

3. From the terminal, run `npm electron-packager.` (I used yarn)
4. Copy the `.app` to the Applications folder
5. Run the app from the launchpad
6. Open safari, type `myapp://` and hit enter in the browserbar, click `Allow`
7. Observe that the `second instance` message is logged in the dev tools console - (this is the unexpected behavior)
8. Close the app (quit in dock)
9. Repeat step 6 (I had to do it twice with this setup to get the logging), notice that the `open-url` message appears as expected

### Screenshots
n/a

### Additional Information
I did note this snippet in the docs, though it didn't provide much insight on how to resolve this: https://electronjs.org/docs/api/app#apprequestsingleinstancelock
```
On macOS, the system enforces single instance automatically when users try to open a second instance of your app in Finder, and the open-file and open-url events will be emitted for that. However when users start your app in command line, the system's single instance mechanism will be bypassed, and you have to use this method to ensure single instance.
```
TLDR; How can I get the protocol URL on a mac when the app is opened/focused using a protocol URL and the `second-isntance` event fires instead of the `open-url` event?

**Comment by phanmn** [comment]
- You should set LSMultipleInstancesProhibited to true in info.plist.
- If you are using electron builder, you can set 'LSMultipleInstancesProhibited: true' at mac.extendInfo For example
```
mac: {
  ...
  extendInfo: {
    LSMultipleInstancesProhibited: true,
  }
}
```

**Comment by electron-triage** [comment]
The Electron version reported on this issue is no longer supported. See our [supported versions documentation](https://www.electronjs.org/docs/tutorial/support#supported-versions).

If this is still reproducible on a supported version, please open a new issue with any other new information that a maintainer should know.

Thank you for taking the time to report this issue and helping to make Electron better! Your help is appreciated.
